src/Servers/Kestrel/Core/src/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -108,32 +108,52 @@ public static string GetAsciiStringNonNullCharacters(this Span<byte> span)
         public static string GetAsciiOrUTF8StringNonNullCharacters(this ReadOnlySpan<byte> span)
             => StringUtilities.GetAsciiOrUTF8StringNonNullCharacters(span, DefaultRequestHeaderEncoding);
 
-        public static string GetRequestHeaderString(this ReadOnlySpan<byte> span, string name, Func<string, Encoding?> encodingSelector)
+        public static string GetRequestHeaderString(this ReadOnlySpan<byte> span, string name, Func<string, Encoding?> encodingSelector, bool checkForNewlineChars = false)
```

Comment:
Since this is internal, can we just require this param? False feels like it might be an unsafe default if we're not careful.

Reply:
Sure, I don't see the downside

Reply:
https://github.com/dotnet/aspnetcore/pull/34441/commits/75812b8f61455b43272f5214975201595cd6634f